R/expr-string.R

```diff
@@ -0,0 +1,1158 @@
+      self$`_rexpr`$str_to_time(format, strict, cache)
+    } else {
+      abort("`dtype` must be of type Date, Datetime, or Time.")
+    }
+  })
+}
+
+#' Convert a String column into a Date column
+#'
+#' @inheritParams expr_str_strptime
+#'
+#' @return [Expr][Expr_class] of Date type
+#' @seealso
+#' - [`<Expr>$str$strptime()`][expr_str_strptime]
+#' @examples
+#' s <- as_polars_series(c("2020/01/01", "2020/02/01", "2020/03/01"))
+#'
+#' s$str$to_date()
+#'
+#' # by default, this errors if some values cannot be converted
+#' s <- as_polars_series(c("2020/01/01", "2020 02 01", "2020-03-01"))
+#' try(s$str$to_date())
+#' s$str$to_date(strict = FALSE)
+expr_str_to_date <- function(format = NULL, ..., strict = TRUE, exact = TRUE, cache = TRUE) {
+  wrap({
+    check_dots_empty0(...)
+    self$`_rexpr`$str_to_date(format, strict, exact, cache)
+  })
+}
+
+#' Convert a String column into a Time column
+#'
+#' @inheritParams expr_str_strptime
+#'
+#' @return [Expr][Expr_class] of Time type
+#' @seealso
+#' - [`<Expr>$str$strptime()`][expr_str_strptime]
+#' @examples
+#' s <- as_polars_series(c("01:00", "02:00", "03:00"))
+#'
+#' s$str$to_time("%H:%M")
+expr_str_to_time <- function(format = NULL, ..., strict = TRUE, cache = TRUE) {
+  wrap({
+    check_dots_empty0(...)
+    self$`_rexpr`$str_to_time(format, strict, cache)
+  })
+}
+
+#' Convert a String column into a Datetime column
+#'
+#' @inheritParams expr_str_strptime
+#' @param time_unit Unit of time for the resulting Datetime column. If `NULL` (default),
+#' the time unit is inferred from the format string if given,
+#' e.g.: `"%F %T%.3f"` => [`pl$Datetime("ms")`][pl_Datetime].
+#' If no fractional second component is found, the default is `"us"` (microsecond).
+#' @param time_zone for the resulting [Datetime][pl_Datetime] column.
+#' @param exact If `TRUE` (default), require an exact format match. If `FALSE`, allow the format to match
+#' anywhere in the target string. Note that using `exact = FALSE` introduces a performance
+#' penalty - cleaning your data beforehand will almost certainly be more performant.
+#' @return [Expr][Expr_class] of [Datetime][pl_Datetime] type
+#' @seealso
+#' - [`<Expr>$str$strptime()`][expr_str_strptime]
+#' @examples
+#' s <- as_polars_series(c("2020-01-01 01:00Z", "2020-01-01 02:00Z"))
+#'
+#' s$str$to_datetime("%Y-%m-%d %H:%M%#z")
+#' s$str$to_datetime(time_unit = "ms")
+expr_str_to_datetime <- function(
+    format = NULL,
+    ...,
+    time_unit = NULL,
+    time_zone = NULL,
+    strict = TRUE,
+    exact = TRUE,
+    cache = TRUE,
+    ambiguous = "raise") {
+  wrap({
+    check_dots_empty0(...)
+    self$`_rexpr`$str_to_datetime(
+      self, format, time_unit, time_zone, strict, exact, cache, ambiguous
+    )
+  })
+}
+
+#' Get the number of bytes in strings
+#' @description
+#' Get length of the strings as UInt32 (as number of bytes). Use `$str$len_chars()`
+#' to get the number of characters.
+#'
+#' @keywords ExprStr
```

Comment:
keywords are needed?

Reply:
Not really, but that's unrelated to the switch of codebase. This can be addressed later IMO.

Reply:
I've removed them as I was doing the other docs changes